src/p2p/Rotation.ts

```diff
@@ -81,14 +81,28 @@ export function updateRecord(
     nestedCountersInstance.countEvent('p2p', `results of getExpiredRemoved: expired: ${expired} removed: ${removed.length}`, 1)
     if (logFlags && logFlags.verbose) console.log(`results of getExpiredRemoved: expired: ${expired} removed: ${removed.length} array: ${removed}`)
   }
-
-  // Allow the autoscale module to set this value
-  const { expired, removed } = getExpiredRemovedV2(prev, lastLoggedCycle, txs, info)
-  nestedCountersInstance.countEvent('p2p', `results of getExpiredRemovedV2: expired: ${expired} removed: ${removed.length}`, 1)
-  if (logFlags && logFlags.verbose) console.log(`results of getExpiredRemovedV2: expired: ${expired} removed: ${removed.length} array: ${removed}`)
-
-  record.expired = expired
-  record.removed = removed // already sorted
+  
+  const problemNodeRemovalEnabled = config.p2p.enableProblematicNodeRemoval && currentCycle >= config.p2p.enableProblematicNodeRemovalOnCycle;
+  // we only want to use the problematic node removal logic if we are past the enableProblematicNodeRemovalOnCycle and have a full history of refutes
+  // note: we may want to wait an additional config.p2p.problematicNodeHistoryLength cycles before we start removing problematic nodes
+  //       this would give us a full history of refutes before we start removing problematic nodes
+  if (problemNodeRemovalEnabled === false) {
+    // Allow the autoscale module to set this value
+    const { expired, removed } = getExpiredRemovedV2(prev, lastLoggedCycle, txs, info)
+    nestedCountersInstance.countEvent('p2p', `results of getExpiredRemovedV2: expired: ${expired} removed: ${removed.length}`, 1)
+    if (logFlags && logFlags.verbose) console.log(`results of getExpiredRemovedV2: expired: ${expired} removed: ${removed.length} array: ${removed}`)
+      
+    record.expired = expired
+    record.removed = removed // already sorted
+  } else {
+    const { expired, removed, problematic } = getExpiredRemovedV3(prev, lastLoggedCycle, txs, info)
+    nestedCountersInstance.countEvent('p2p', `results of getExpiredRemovedV2: expired: ${expired} removed: ${removed.length} problematic: ${problematic}`, 1)
+    if (logFlags && logFlags.verbose) console.log(`results of getExpiredRemovedV2: expired: ${expired} removed: ${removed.length} array: ${removed} problematic: ${problematic}`)
```

Comment:
if  you replace:
if (logFlags && logFlags.verbose) console.log(  <the message>
with:
/* prettier-ignore */ if(logFlags?.node_rotation_debug) logger.mainLog_debug('UNIQUE_TAG_HERE', <the message>

it will make evaluating this and other rotation related systems much easier on busy production nodes.
enabled with:
http://localhost:9001/log-flag?name=node_rotation_debug&value=true
(or by the ui in the debug app.. I still need to merge this flag to debug app dev)

Reply:
on a large, live, busy network the spam of turning on verbose is quite high.   The above helped me validate all my recent work to rotation etc.

Reply:
also this will mean the context of this system will be linked to my other recent logs